app/controllers/locations/map_controller.rb

```diff
@@ -0,0 +1,27 @@
+require_dependency "locations/application_controller"
+
+module Locations
+  class MapController < ApplicationController
+
+    def index
+        @map = Map.new(id: map_params[:id],loc: map_params[:loc])
+
+        if !@map.url.nil?
+          if @map.url.include? "http"
+            redirect_to @map.url
+          else
+            render plain: "#{@map.url}", status: 200
```

Comment:
In this case the result is a message about what do. Ultimately I think we should try and provide a better landing page for that than just a plain text string. Slightly odd to have variable with url that is sometimes just a string.